libs/ui/design-system/src/lib/switch.tsx

```diff
@@ -1 +1,39 @@
-export { Switch } from '@nmit-coursition/ui/primitives/switch'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+import { Label } from '@nmit-coursition/ui/primitives'
+import * as SwitchPrimitive from '@nmit-coursition/ui/primitives/switch'
+import { cn } from '@nmit-coursition/ui/utils'
+
+/** todo: import util from `typescript` file when mixins architecure is merged  */
+type OverrideProps<T, V> = V & Omit<T, keyof V>
+
+const switchVariants = cva('', {
+  variants: {
+    variant: {
+      /**
+       * OFF: bg-#4A044E, thumb-#FEF3C7
+       * ON: bg-[#FEF3C7], thumb-[#4A044E]
+       */
+      default: '',
+      disabled: '',
+      outlined: '', // default variant with an outline
+      critical: '',
+    },
+  },
+  defaultVariants: {
+    variant: 'default',
+  },
+})
+
+interface Props extends OverrideProps<SwitchPrimitive.RootProps, { id: string }>, VariantProps<typeof switchVariants> {
+  label: string
```

Comment:
**thought:** I am thinking with the designer, that we might need also `subtext`, but that will be simple addition.

Also I think we need `name` just like other inputs, because we need to use it from `FormData`.

Reply:
Good idea, implementing that now..